Add support for window feature - Danfoss Ally

robertalexa opened the pull request:
This adds support for the new feature included with firmware 1.18 which allows turning the window feature on and off.

Not supported by Hive TRV - or at least the version of firmware I have - don't even know if there is a newer one as I have not connected them to Hive in a looong time.

Not sure about Popp TRV, hence me not adding it. I am sure someone will try it on the Popp and do a PR if they find it works.

Confirmed working on Danfoss Ally 1.18 in this post https://github.com/Koenkk/zigbee2mqtt/issues/4504#issuecomment-1028116201

robertalexa commented on devices/danfoss.js:
```diff
@@ -53,6 +54,8 @@ module.exports = [
             exposes.binary('radiator_covered', ea.ALL, true, false)
                 .withDescription('Set if the TRV should solely rely on external_measured_room_sensor or operate in offset mode. ' +
                     '`false` = Auto Offset Mode or `true` = Room Sensor Mode'),
+            exposes.binary('window_open_feature_enable', ea.ALL, true, false)
```
I have named it so purely to match zigbee-herdsman identically (which was PRed by someone else)

https://github.com/Koenkk/zigbee-herdsman/blob/44f479d7c82ce9ad9a85a6c3cf1d0aaf216341ca/src/zcl/definition/cluster.ts#L1994

I can change just the attribute and leave the property as `danfossWindowOpenFeatureEnable`

Koenkk replied:
Sorry to bother again, but `enabled` seems redundant, so `window_open_feature` might even be better.

robertalexa replied:
Done :)

robertalexa replied:
If you want me to i can also do a PR on z-h and rename the property, and do another commit here to match it?

Koenkk replied:
ZH can be kept as is since users are not bothered with this.

sjorge commented:
We should also read danfossWindowOpenFeatureEnable attribute in the configure function so we populate the value after pairing, we already do this for other attributes.

robertalexa commented:
@sjorge Eagle eyed :) Good spot. I have now added that. I have added this attribute blindly and had no visual trigger. And obviously my mind didn't flag it either :D
